2-api/cyf-ecommerce-api/server.js

```diff
@@ -60,42 +53,140 @@ app.post("/customers", function (req, res) {
     });
 });
 
-// GET ALL THE SUPPLIERS
-app.get("/suppliers", function (req, res) {
+// ADDING A NEW PRODUCT
+
+app.post("/products", function (req, res) {
+  let productName = req.body.product_name;
   pool
-    .query("SELECT * FROM suppliers")
+    .query(`SELECT * FROM products WHERE product_name = $1`, [productName])
+    .then((result) => {
+      if (result.rows.length > 0) {
+        return res.status(400).send({
+          msg: `A product with similar name:${productName} already exists !`,
+        });
+      } else {
+        let query = `INSERT INTO products (product_name) VALUES($1)`;
+        let params = [productName];
+        pool
+          .query(query, params)
+          .then(() =>
+            res.send({ msg: `Product ${productName} added successfully ` })
+          )
+          .catch((error) => {
+            console.error(error);
+            res.status(500).json(error);
+          });
+      }
+    });
+});
+
+//  ADD PRODUCT-AVAILABILITY
+//  NOT working properly
+app.post("/availability", function (req, res) {
+  let productId = req.body.prodId;
+  let productPrice = req.body.unitPrice;
+  let supplierId = req.body.suppId;
+
+  if (!Number(productPrice) || productPrice < 0 || !productId || !supplierId) {
```

Comment:
from personal experience, using !poductId.length and !supplierId.length avoids certain errors